lib/src/main/java/io/github/taz03/jia/InstagramClient.java

```diff
@@ -49,20 +56,40 @@ public final class InstagramClient {
      * @param password The instagram password corresponding to the username provided
      */
     public InstagramClient(String username, String password) {
-        this(username, password, null);
+        this.username = username;
+        this.password = password;
+    }
+
+    public InstagramClient(String user,String pass,String id,String agent){
+        this.username = user;
+        this.password = pass;
+        this.userAgent = agent;
+        this.androidId = id;
+    }
+
+    private void configureDeviceIds() throws MalformedURLException, IOException {
+        var id = UUID.randomUUID().toString().replaceAll("-", "").toUpperCase();
+        this.androidId += "android-" + id.substring(id.length() - 16);
+        
+        var random = new Random().nextInt(1783);
+        var scanner = new Scanner(new URL("https://raw.githubusercontent.com/Taz03/JIA/main/user-agents.txt").openStream());
+        for (int i = 0; i < random; i++) {
+            scanner.skip("");
+            userAgent = scanner.nextLine();
+        }
     }
 
     /**
      * Makes an Instagram client logined.
-     *
-     * @param username      The instagram username
-     * @param password      The instagram password corresponding to the username provided
+     * @param androidId The andorid id that was used in intial login
+     * @param userAgent The user agent string that was used while logging initally.
      * @param authorization The authorization token to corresponding to the username and password provided
      */
-    public InstagramClient(String username, String password, String authorization) {
-        this.username = username;
-        this.password = password;
+    public InstagramClient(String username,String authorization,String userAgent) {
```

Comment:
wrong formatting

Reply:
Just merge and then commit it by yourself